doc/Command-Reference.md

```diff
@@ -3763,39 +3762,82 @@ This command displays all the mirror sessions that are configured.
 
 - Example:
   ```
-  admin@sonic:~$ show mirror session
-  Name       Status    SRC IP     DST IP    GRE    DSCP    TTL    Queue
-  ---------  --------  ---------  --------  -----  ------  -----  -------
+  admin@sonic:~$ show mirror_session
+  ERSPAN Sessions
+  Name       Status    SRC IP     DST IP    GRE    DSCP    TTL    Queue    Policer    Monitor Port    SRC Port    Direction
+  ------     --------  --------   --------  -----  ------  -----  -------  ---------  --------------  ----------  -----------
   everflow0  active    10.1.0.32  10.0.0.7
+
+  SPAN Sessions
+  Name    Status    DST Port    SRC Port       Direction
+  ------  --------  ----------  -------------  -----------
+  port0   active    Ethernet0   PortChannel10  rx
   ```
 
 ### Mirroring Config commands
 
 **config mirror_session**
 
 This command is used to add or remove mirroring sessions. Mirror session is identified by "session_name".
-While adding a new session, users need to configure the following fields that are used while forwarding the mirrored packets.
+This command supports configuring both SPAN/ERSPAN sessions.
+In SPAN user can configure mirroring of list of source ports/LAG to destination port in ingress/egress/both directions.
+In ERSPAN user can configure mirroring of list of source ports/LAG to a destination IP.
+Both SPAN/ERSPAN support ACL based mirroring and can be used in ACL configurations.
+
+While adding a new ERSPAN session, users need to configure the following fields that are used while forwarding the mirrored packets.
 
 1) source IP address,
 2) destination IP address,
 3) DSCP (QoS) value with which mirrored packets are forwarded
 4) TTL value
 5) optional - GRE Type in case if user wants to send the packet via GRE tunnel. GRE type could be anything; it could also be left as empty; by default, it is 0x8949 for Mellanox; and 0x88be for the rest of the chips.
 6) optional - Queue in which packets shall be sent out of the device. Valid values 0 to 7 for most of the devices. Users need to know their device and the number of queues supported in that device.
+7) optional - Policer which will be used to control the rate at which frames are mirrored.
+8) optional - List of source ports which can have both Ethernet and LAG ports.
+9) optional - Direction - Mirror session direction when configured along with Source port. (Supported rx/tx/both. default direction is both)
 
 - Usage:
   ```
-  config mirror_session add <session_name> <src_ip> <dst_ip> <dscp> <ttl> [gre_type] [queue]
+  config mirror_session add erspan <session_name> <src_ip> <dst_ip> <dscp> <ttl> [gre_type] [queue] [policer <policer_name>] [source-port-list] [direction]
```

Comment:
Is there any way we can default to `erspan` if nothing is specified here? The reason I ask is because this might present backwards compatibility issues, especially for users that have some devices with newer OS versions and some with older OS versions.

Reply:
yes. true.  so we have the default mirror type as ERSPAN in swss codes.
I can think of below options for this CLI.
config mirror_session span add <session_name> 
config mirror_session add <session_name> ( will default to ERSPAN).

@daall Is this ok ?  or u have any suggestions ?

Reply:
That sounds OK to me. @jleveque do you have any concerns with that approach or alternate suggestions?

Reply:
I prefer keeping the syntax unified as it's easier to understand and figure out, e.g.,

```
config mirror_session add span ...
config mirror_session add erspan ...
```

However, if backward compatibility is a concern, we should accept both. If no mirror type is specified, default to ERSPAN. This will also need to be documented well.

Reply:
Thanks @jleveque @daall . I will modify as below.
```
config mirror_session span add ...
config mirror_session erspan add ...
config mirror_session add ... (defaults to erspan. to be deprecated)
```

Thanks

Reply:
Modified the CLI as discussed.